File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int) // val-> index

    for i,n := range nums{
        complement := target - nums[i]
        if idx,ok := seen[complement]; ok{ //map[key] returns 2 values
            return []int{idx,i}
        }
        seen[n] = i 
    }
    return []int{}
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(len(nums)):
            hashmap[nums[i]] = i
        for j in range(len(nums)):
            complement = target - nums[j]
            if complement in hashmap and hashmap[complement] != j:
                return [j, hashmap[complement]]
        return []
